consent/helper_test.go

```diff
@@ -95,3 +99,264 @@ func TestMatchScopes(t *testing.T) {
+		csrfValue string
+		sameSite  http.SameSite
+	}
+	for k, tc := range []struct {
+		cookies                  []cookie
+		csrfValue                string
+		sameSiteLegacyWorkaround bool
+		expectError              bool
+	}{
+		{
+			cookies:                  []cookie{},
+			csrfValue:                "CSRF-VALUE",
+			sameSiteLegacyWorkaround: false,
+			expectError:              true,
+		},
+		{
+			cookies:                  []cookie{},
+			csrfValue:                "CSRF-VALUE",
+			sameSiteLegacyWorkaround: true,
+			expectError:              true,
+		},
+		{
+			cookies: []cookie{
+				{
+					name:      cookieAuthenticationCSRFName,
+					csrfValue: "WRONG-CSRF-VALUE",
+					sameSite:  http.SameSiteDefaultMode,
+				},
+			},
+			csrfValue:                "CSRF-VALUE",
+			sameSiteLegacyWorkaround: false,
+			expectError:              true,
+		},
+		{
+			cookies: []cookie{
+				{
+					name:      cookieAuthenticationCSRFName,
+					csrfValue: "WRONG-CSRF-VALUE",
+					sameSite:  http.SameSiteDefaultMode,
+				},
+			},
+			csrfValue:                "CSRF-VALUE",
+			sameSiteLegacyWorkaround: true,
+			expectError:              true,
+		},
+		{
+			cookies: []cookie{
+				{
+					name:      cookieAuthenticationCSRFName,
+					csrfValue: "CSRF-VALUE",
+					sameSite:  http.SameSiteDefaultMode,
+				},
+			},
+			csrfValue:                "CSRF-VALUE",
+			sameSiteLegacyWorkaround: false,
+			expectError:              false,
+		},
+		{
+			cookies: []cookie{
+				{
+					name:      cookieAuthenticationCSRFName,
+					csrfValue: "CSRF-VALUE",
+					sameSite:  http.SameSiteDefaultMode,
+				},
+			},
+			csrfValue:                "CSRF-VALUE",
+			sameSiteLegacyWorkaround: true,
+			expectError:              false,
+		},
+		{
+			cookies: []cookie{
+				{
+					name:      legacyCsrfSessionName(cookieAuthenticationCSRFName),
+					csrfValue: "CSRF-VALUE",
+					sameSite:  http.SameSiteDefaultMode,
+				},
+			},
+			csrfValue:                "CSRF-VALUE",
+			sameSiteLegacyWorkaround: false,
+			expectError:              true,
+		},
+		{
+			cookies: []cookie{
+				{
+					name:      legacyCsrfSessionName(cookieAuthenticationCSRFName),
+					csrfValue: "CSRF-VALUE",
+					sameSite:  http.SameSiteDefaultMode,
+				},
+			},
+			csrfValue:                "CSRF-VALUE",
+			sameSiteLegacyWorkaround: true,
+			expectError:              false,
+		},
+		{
+			cookies: []cookie{
+				{
+					name:      cookieAuthenticationCSRFName,
+					csrfValue: "CSRF-VALUE",
+					sameSite:  http.SameSiteDefaultMode,
+				},
+				{
+					name:      legacyCsrfSessionName(cookieAuthenticationCSRFName),
+					csrfValue: "CSRF-VALUE",
+					sameSite:  http.SameSiteDefaultMode,
+				},
+			},
+			csrfValue:                "CSRF-VALUE",
+			sameSiteLegacyWorkaround: false,
+			expectError:              false,
+		},
+		{
+			cookies: []cookie{
+				{
+					name:      cookieAuthenticationCSRFName,
+					csrfValue: "CSRF-VALUE",
+					sameSite:  http.SameSiteDefaultMode,
```

Comment:
Shouldn't this be `http.SameSiteNoneMode`?

Reply:
Yup, will update.